states/team.py

```diff
@@ -1,7 +1,35 @@
+from typing import Optional
 
+from config.config import TEXT
+from states.battle import Battle
+from states.state import State, StateOptions
 
-class Team:
+
+class TeamOptions(StateOptions):
+    """Enum values for options availabe in team menu"""
+
+    TOSSPKMN = 1
+    REORDER = 2
+    EXIT = 3
+
+
+class Team(State):
     """Game state for player team management"""
 
+    option: Optional[TeamOptions] = None
+
     def __init__(self):
         super().__init__()
+        while self.option is not TeamOptions.EXIT:
+            print(TEXT["TEAM"]["ENTRY"])
+            self.option = self.check_input(TeamOptions)
+            self.choose_option()
+
+    def choose_option(self):
+        """Enters next game state based on player choice"""
+        if self.option == TeamOptions.TOSSPKMN:
+            Battle()
+        elif self.option == TeamOptions.REORDER:
+            Team()
+        elif self.option == TeamOptions.EXIT:
+            print(TEXT["TEAM"]["EXIT"])
```

Comment:
I don't think anything needs to be done on `TeamOptions.EXIT`. The game will already print to the player that they chose the choice 'exit'. By default, `choose_option` should end if the player did not choose one of `TeamOptions` that satisfies the other `if` conditions, so the `Team` State should exit to Menu by default. In other words, there's no need for this if statement, or the `EXIT` text on `TEAM`.